src/system/Nav/NavItem.tsx

```diff
@@ -38,52 +41,91 @@ const NavItemBase = forwardRef< HTMLLIElement, NavItemBaseProps >(
 	)
 );
 
-const NavLink = forwardRef< HTMLAnchorElement, NavItemProps >(
-	(
-		{ children, href, active, disabled, variant = 'primary', ...props }: NavItemProps,
-		ref: Ref< HTMLAnchorElement >
-	) => (
-		<NavigationMenu.Link
-			className={ classNames( `${ VIP_NAV }-item-link` ) }
-			href={ href }
-			ref={ ref }
-			sx={ navItemStyles( variant ) }
-			active={ active }
-			aria-current={ active ? 'page' : undefined }
-			aria-disabled={ disabled }
-			{ ...props }
-		>
-			{ children }
-		</NavigationMenu.Link>
-	)
-);
-
 export interface NavItemProps extends NavigationMenu.NavigationMenuLinkProps {
 	className?: string;
 	disabled?: boolean;
 	variant?: NavVariant;
+	icon?: JSX.Element;
```

Comment:
Should `icon` also be a component prop (`Icon`)? Should the size (and maybe even icon type) be passed by this component as props? I don't really know what our long-term icon strategy is. Should this repo export an Icon component that provides all possible icon types?

Reply:
@chriszarate I believe the way it is now gives more flexibility. The menu is dynamic so that it would not be possible to predict all icons.

I think we could use the IconComponent property to keep the same pattern we are using now for the LinkComponent. I will update the code accordingly.

Reply:
@chriszarate I thought a little more about the subject and I went with `renderIcon` property passing `size` to the consumer. TypeScript cannot understand what's coming from IconComponent, especially when it's a customized component, like react-icon. To be safe, a `renderIcon` would work, and still be flexible.